pkg/services/live/push/push.go

```diff
@@ -46,7 +68,7 @@ func (g *Gateway) Init() error {
 // Run Gateway.
 func (g *Gateway) Run(ctx context.Context) error {
 	if !g.IsEnabled() {
-		logger.Debug("GrafanaLive feature not enabled, skipping initialization of Telemetry Gateway")
+		logger.Debug("GrafanaLive feature not enabled, skipping initialization of Live Push Gateway")
```

Comment:
If you're not planning to do any background processing in here you can just skip implementing `Run`. See https://github.com/grafana/grafana/blob/438b403acc3e4a374872104a2ef1a510abdd5075/pkg/registry/registry.go#L123-L130

Reply:
Thanks, maybe this should be useful for stopping cache invalidation routine.

Reply:
Maybe. But maybe easier to handle it during the request?

Reply:
What to you mean? The main latency reduction here is removing token check on each request. I was planning to add a goroutine that will periodically invalidate/remove cached keys. Doing this during request will result into the same 10+ ms latency. Or you suggest sth else?

Reply:
Related to my https://github.com/grafana/grafana/pull/33212#discussion_r617698530 where I suggest storing the time when the token expires in the cache and then doing the expiration check in the request shouldn't add any latency. But would you like to do it in the background instead that should also work. However might be a risk of more contention/locking of the mutex?

Reply:
Ah, right - I see now, we have a time of token expiration. Cool - of course better to just check expiration during a request then.

Reply:
Hmm, though can be revoked - since tokens long lived usually this can be a problem

Reply:
So I suppose we need to to both: invalidate periodically and check expiration on each request for existing token.

Reply:
Yes you're right. Didn't think of that :+1: 

Another thing that I just came to think about is in the case you have many channels/streams data being pushed to, there's a chance of contention/locking. If that's a potential problem you might want to include the channel/stream in the cache key identifier. But maybe something to consider later.

Reply:
Also: looks like there could be tokens without expiration since `Expires` is nullable:

```go
type ApiKey struct {
        ...
	Expires *int64
}
```

Reply:
> If that's a potential problem you might want to include the channel/stream in the cache key identifier. But maybe something to consider later.

Yep, this is definitely an advice for a future

Reply:
Yes, there could be tokens without expiration

Reply:
what about using a library like:
https://github.com/patrickmn/go-cache

This has builtin support to the periodic cleanup and checking individual entries when you get them